src/test/fakes.rs

```diff
@@ -76,15 +80,40 @@ impl<'db> FakeRelease<'db> {
         self
     }
 
+    pub(crate) fn rustdoc_file<P, D>(mut self, path: P, data: D) -> Self
+        where P: Into<String>,
+              D: Into<Vec<u8>>,
+        {
+        let (path, data) = (path.into(), data.into());
+        self.rustdoc_files.push((path, data));
+
+        self
+    }
+
     pub(crate) fn create(self) -> Result<i32, Error> {
         let tempdir = tempdir::TempDir::new("docs.rs-fake")?;
 
+        let upload_files = |prefix, files: Vec<(String, Vec<u8>)>, package: &MetadataPackage, db: &TestDatabase| {
+            for (path, data) in files {
+                let file = tempdir.path().join(&path);
+                std::fs::write(file, data)?;
```

Comment:
I just realized there is a bug here, the files from the first call will also be uploaded on the second call.